### PR TITLE
upgrade to support spree v4.5

### PR DESCRIPTION
**Pull request by kimsrung**
- update to support spree 4.5
- update to support ruby 3.2
- fix test

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,8 +5,10 @@ git_source(:github) do |repo_name|
   "https://github.com/#{repo_name}.git"
 end
 
-gem 'spree', github: 'spree/spree', branch: 'master'
-gem 'spree_auth_devise', github: 'spree/spree_auth_devise', branch: 'master'
+gem 'spree', github: 'spree/spree', branch: 'main'
```
Comment (channainfo) [inline Gemfile]: let's point to the specific 4.5 instead of the main branch.

File: Gemfile [hunk 2]
```diff
@@ -5,8 +5,10 @@ git_source(:github) do |repo_name|
...
+gem 'spree_auth_devise', github: 'spree/spree_auth_devise', branch: 'main'
 gem 'rails-controller-testing'
+gem 'byebug'
```
Comment (channainfo) [inline Gemfile]: No need byebug

File: spec/features/admin_spec.rb [hunk 3]
```diff
@@ -19,6 +19,7 @@
       within("tr#review_#{review.id}") do
         find('.approve').click
       end
+      sleep(3)
```
Comment (channainfo) [inline spec/features/admin_spec.rb]: Remove this.

**Review by channainfo — CHANGES_REQUESTED** [review]

File: spec/features/admin_spec.rb (hunk 3 above)
Comment (kimsrung) [reply spec/features/admin_spec.rb]: if we removed this, the spec would fail due to the delay of response.

**Pull request merged**